test/spec/modules/timeoutRtdProvider_spec.js

```diff
@@ -0,0 +1,316 @@
+    expect(timeoutSubmodule.init()).to.equal(true);
+  });
+
+  it('should make a request to the endpoint url if it is provided, and handle the response', () => {
+    const response = '{"deviceType":{ "2": 50, "4": 100, "5": 200 }}'
+    const ajaxStub = sandbox.stub().callsFake(function (url, callbackObj) {
+      callbackObj.success(response);
+    });
+    sandbox.stub(ajax, 'ajaxBuilder').callsFake(function () { return ajaxStub });
+
+    const reqBidsConfigObj = {}
+    const expectedLink = 'https://somelink.json'
+    const config = {
+      'name': 'timeout',
+      'params': {
+        'endpoint': {
+          url: expectedLink
+        }
+      }
+    }
+    const handleTimeoutIncrementStub = sandbox.stub(timeoutRtdFunctions, 'handleTimeoutIncrement');
+    timeoutSubmodule.getBidRequestData(reqBidsConfigObj, function() {}, config)
+
+    expect(ajaxStub.calledWith(expectedLink)).to.be.true;
+    expect(handleTimeoutIncrementStub.calledWith(reqBidsConfigObj, JSON.parse(response))).to.be.true;
+  });
+
+  it('should make a request to the endpoint url and ignore the rules object if the endpoint is provided', () => {
+    const ajaxStub = sandbox.stub().callsFake((url, callbackObj) => {});
+    sandbox.stub(ajax, 'ajaxBuilder').callsFake(() => ajaxStub);
+    const expectedLink = 'https://somelink.json'
+    const config = {
+      'name': 'timeout',
+      'params': {
+        'endpoint': {
+          url: expectedLink
+        },
+        'rules': {
+          'includesVideo': {
+            'true': 200,
+          },
+        }
+      }
+    }
+    timeoutSubmodule.getBidRequestData({}, function() {}, config);
+    expect(ajaxStub.calledWith(expectedLink)).to.be.true;
+  });
+
+  it('should use the rules object if there is no endpoint url', () => {
+    const config = {
+      'name': 'timeout',
+      'params': {
+        'rules': {
+          'includesVideo': {
+            'true': 200,
+          },
+        }
+      }
+    }
+    const handleTimeoutIncrementStub = sandbox.stub(timeoutRtdFunctions, 'handleTimeoutIncrement');
+    const reqBidsConfigObj = {};
+    timeoutSubmodule.getBidRequestData(reqBidsConfigObj, function() {}, config);
+    expect(handleTimeoutIncrementStub.calledWith(reqBidsConfigObj, config.params.rules)).to.be.true;
+  });
+
+  it('should exit quietly if no relevant timeout config is found', () => {
+    const callback = sandbox.stub()
+    const ajaxStub = sandbox.stub().callsFake((url, callbackObj) => {});
+    sandbox.stub(ajax, 'ajaxBuilder').callsFake(function() { return ajaxStub });
+    const handleTimeoutIncrementStub = sandbox.stub(timeoutRtdFunctions, 'handleTimeoutIncrement');
+
+    timeoutSubmodule.getBidRequestData({}, callback, {});
+
+    expect(handleTimeoutIncrementStub.called).to.be.false;
+    expect(callback.called).to.be.true;
+    expect(ajaxStub.called).to.be.false;
+  });
+
+  it('should be able to increment the timeout with the calculated timeout modifier', () => {
```

Comment:
can you add a unit test for the multiple rules match case?

Reply:
This should be handled by this test case: https://github.com/prebid/Prebid.js/pull/7395/files#diff-49900727474e17b1862be0b57a305580c07d69ca22e2e4c924a78d0630631735R173-R205, though I added one anyway, since this new one also checks if the timeout is incremented properly when there's multiple matching rules